src/ctapipe/calib/camera/tests/test_extractor.py

```diff
@@ -0,0 +1,136 @@
+"""
+Tests for StatisticsExtractor and related functions
+"""
+
+import numpy as np
+import pytest
+from astropy.table import Table
+from astropy.time import Time
+
+from ctapipe.calib.camera.extractor import PlainExtractor, SigmaClippingExtractor
+
+
+def test_extractors(example_subarray):
+    """test basic functionality of the StatisticsExtractors"""
+
+    # Create dummy data for testing
+    times = Time(
+        np.linspace(60117.911, 60117.9258, num=5000), scale="tai", format="mjd"
+    )
+    event_ids = np.linspace(35, 725000, num=5000, dtype=int)
+    ped_data = np.random.normal(2.0, 5.0, size=(5000, 2, 1855))
```

Comment:
Please use a fixed random seed so we don't get spurious test failures and use the new numpy random API, not the deprecated one:

```
rng = np.random.default_rng(0)
ped_data = rng.normal(...)
charge_data = rng.normal(...)
```